File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.java
class Solution {
    public int waysToSplitArray(int[] nums) {
                long  [] left = new long  [nums.length];
        long  lSum = 0;
        for (int i = 0; i<nums.length; i++) {
            lSum += nums[i];
            left[i] = lSum;
        }
        System.out.println(Arrays.toString(left));
        long  [] right = new long[nums.length];
        long  rSum = 0;
        for (int j = nums.length-1; j>=0; j--) {
            rSum += nums[j];
            right[j] = rSum;
        }
        System.out.println(Arrays.toString(right));
        int i = 0, j = 1;
        int ans = 0;
        while (i<left.length && j<right.length) {
            if (left[i]>=right[j]) ans++;
            i++;
            j++;
        }
        return ans;
    }
}

File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.go
func waysToSplitArray(nums []int) int {
    total := 0
    for i := 0; i < len(nums); i++ {
        total += nums[i]
    }

    count := 0
    runningLeftSum := 0
    for i := 0; i < len(nums)-1; i++ {
        runningLeftSum += nums[i]
        rightSum := total-runningLeftSum
        if runningLeftSum >= rightSum {
            count++
        }
    }
    return count
}
